tasks/preparation-provisioning.yaml

```
@@ -0,0 +1,95 @@
+   - name: CUSTOM_IMAGE
+     type: string
+     description: The container image to use for the provisioning setup task
+     default: quay.io/envision/provisioning-cicd:latest
+   - name: NS_REQUESTER
+     type: string
+     description: The name of the person/pipeline that requested the namespace
+   - name: EPHEMERAL_ENV_PROVIDER_SECRET
+     type: string
+     default: ephemeral-env-provider
+     description: "Secret for connecting to ephemeral env provider cluster"
+   - name: AWS_ACCOUNT_ID_SECRET
+     type: string
+     description: Secret containing the AWS Account ID
+     default: provisioning-aws-account-id
+   - name: AZURE_TENANT_ID_SECRET
+     type: string
+     description: Secret containing the AZURE Tenant ID
+     default: provisioning-azure-tenant-id
+   - name: AZURE_SUBSCRIPTION_ID_SECRET
+     type: string
+     description: Secret containing the AZURE Subscription ID
+     default: provisioning-azure-subscription-id
+   - name: AZURE_RESOURCE_GROUP_SECRET
+     type: string
+     description: Secret containing the AZURE Resource Group
+     default: provisioning-azure-resource-group
+  steps:
+    - name: provisioning-setup
+      image: "$(params.CUSTOM_IMAGE)"
+      env:
+        - name: OC_LOGIN_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.EPHEMERAL_ENV_PROVIDER_SECRET)
+              key: token
+        - name: OC_LOGIN_SERVER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.EPHEMERAL_ENV_PROVIDER_SECRET)
+              key: url
+        - name: AWS_ACCOUNT_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.AWS_ACCOUNT_ID_SECRET)
+              key: AWS_ACCOUNT_ID
+        - name: AZURE_TENANT_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.AZURE_TENANT_ID_SECRET)
+              key: AZURE_TENANT_ID
+        - name: AZURE_SUBSCRIPTION_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.AZURE_SUBSCRIPTION_ID_SECRET)
+              key: AZURE_SUBSCRIPTION_ID
+        - name: AZURE_RESOURCE_GROUP
+          valueFrom:
+            secretKeyRef:
+              name: $(params.AZURE_RESOURCE_GROUP_SECRET)
+              key: AZURE_RESOURCE_GROUP       
+        - name: BONFIRE_BOT
+          value: "true"
+      script: |
+        #!/bin/bash
+        set -ex
+
+        echo "Running login.sh script"
+        main() {
+        local server="${OC_LOGIN_SERVER:?OC_LOGIN_SERVER env var was not defined}"
+        local token="${OC_LOGIN_TOKEN:?OC_LOGIN_SERVER env var was not defined}"
+
+        oc login --token="$token" --server="$server"
+        }
+
+        if [[ "${BASH_SOURCE[0]}" == "$0" ]]; then
+        main "$@"
+        fi
+
+        echo "Running images_db_stub.sh"
+        /images_db_stub.sh \
+          "$AWS_ACCOUNT_ID" \
+          "$AZURE_TENANT_ID" \
+          "$AZURE_SUBSCRIPTION_ID" \
+          "$AZURE_RESOURCE_GROUP"
```

I think this won't work, the script doesn't accept parameters. It expects it as env vars. I assume those are already exported, but if not, we just need to export them:
```suggestion
        export AWS_ACCOUNT_ID
        export AZURE_TENANT_ID
        export AZURE_SUBSCRIPTION_ID
        export AZURE_RESOURCE_GROUP
        
        /images_db_stub.sh 
```